avocado/utils/dmesg.py

```diff
@@ -0,0 +1,111 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import logging
+import os
+import tempfile
+import time
+
+from . import genio, process
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+class DmesgError(Exception):
+    """Base Exception Class for all dmesg  utils exceptions"""
+
+
+def clear_dmesg():
+    """This function clear dmesg
+
+    dmesg operation is privilege user task . This Function need sudo
+    permissions enabled on the target host
+    """
+    cmd = "dmesg -c"
+    status = process.system(cmd, timeout=30, ignore_status=True,
+                            verbose=False, shell=True, sudo=True)
+    if status:
+        raise DmesgError(
+            "Unable to clear dmesg as some issue while clearing")
+
+
+def collect_dmesg(output_file=None):
+    """This Function collect dmesg and save in file
+
+    dmesg operation is privilege user task . This Function need
+    sudo permissions enabled on the target host
+
+    :param output_file : File use for save dmesg output if not provided it use
+                        tmp file which located in system  /tmp path
+    :return: file which contain dmesg
+    """
+    if output_file is None:
+        output_file = tempfile.mkstemp(
+            suffix=".-%s" % time.strftime("%Y-%m-%d:%H:%M:%S"), dir=tempfile.gettempdir())[1]
```

Comment:
It would be better if this is spaced as
```python
        output_file = tempfile.mkstemp(suffix=".-%s" % time.strftime("%Y-%m-%d:%H:%M:%S"),
                                                          dir=tempfile.gettempdir())
```
with the indexing possibly avoided for the sake of `_, output_file = tempfile.mkstemp(...)`.

Reply:
ah it happened by auto-pep8 utility , i will take care